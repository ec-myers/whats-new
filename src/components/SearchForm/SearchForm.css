.searchForm {
  align-items: center;
  display: flex;
  height: 10%;
  justify-content: space-between;
  width: 100%;
}

.searchInput {
  border-radius: 4px;
  border: 2px solid black;
  font-size: 18px;
  height: 50%;
  margin-left: 30px;
  padding-left: 10px;
  width: 50%;
}

.searchButton {
  background-color: papayawhip;
  border-radius: 4px;
  border: 2px solid black;
  font-size: 18px;
  height: 50%;
  margin-right: 30px;
  width: 30%;
}