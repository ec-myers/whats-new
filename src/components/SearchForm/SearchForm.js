import React, { Component } from 'react';
import './SearchForm.css';

class SearchForm extends Component {
  constructor() {
    super();
    this.state = {
      searchInput: ""
    }
  }

  handleChange = (event) => {
    this.setState({ searchInput: event.target.value });
  }
  
  handleClick = (event) => {
    event.preventDefault();
    const searchText = this.state.searchInput.toLowerCase();
    this.props.displaySearchResults(searchText);
    this.setState({ searchInput: "" });
  }

  render() {
    return (
      <form className="searchForm">
        <input 
          className="searchInput" 
          name="searchInput" 
          type="text" 
          placeholder="Looking for something?" 
          value={ this.state.searchInput } 
          onChange={event => this.handleChange(event)}
        />       
        <button 
          className="searchButton" 
          onClick={event => this.handleClick(event)}>Search
        </button>
      </form>
    )
  }
}


export default SearchForm;