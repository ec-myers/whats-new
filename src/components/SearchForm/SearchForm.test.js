import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';

describe ('SearchForm', () => {
  let wrapper;
  const mockDisplaySearch = jest.fn();
  let searchEvent = {target: {name: 'searchInput', value: 'iOS'}}

  beforeEach(() => {
    wrapper = shallow(<SearchForm displaySearchResults={mockDisplaySearch} />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update searchInput in state when change is detected', () => {
    wrapper.instance().handleChange(searchEvent);
    expect(wrapper.state('searchInput')).toEqual('iOS');
  });

  it('should display articles that match the headline search when clicked', () => {
    wrapper.find('button').simulate('click');
    expect(mockDisplaySearch).toHaveBeenCalled();
  });

  it('should reset state after form is cleared', () => {
    wrapper.instance().setState({searchInput:'Pikachu'});
    wrapper.instance().handleClick();
    expect(wrapper.state()).toEqual({searchInput: ''});
  });
});