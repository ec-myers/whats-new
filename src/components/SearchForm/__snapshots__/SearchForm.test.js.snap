// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot 1`] = `
<form
  className="searchForm"
>
  <input
    className="searchInput"
    name="searchInput"
    onChange={[Function]}
    placeholder="Looking for something?"
    type="text"
    value=""
  />
  <button
    className="searchButton"
    onClick={[Function]}
    type="button"
  >
    Search
  </button>
</form>
`;
